password.py

```diff
@@ -1,49 +1,302 @@
 # Password Creator/Manager
 
+import random
+import shane_char 
+import re
+
+account_dict = {}
+account_info_dict = {}
 
 #<---- FIRST MAJOR FUNCTION ---->
 # Creates an account
 def account_create():
 
-    return
+    print("-"*18, "\nAccount Creation!"), print("-"*18)
+
+    while True:
+        username = input("Enter a Username:\n")
+        if len(username) < 5:
+            print("Username has to be 5 characters or more.")
+            continue
+        else:
+            account_info_dict["Username"] = username
+            break
+
+    if password_randomizer_query() == 'no':
+        while True:
+            password = input("Enter a Password:\n")
+            if len(password) < 6:
+                print("Password has to be 6 characters or more.")
+                continue
+            else:
+                account_info_dict["Password"] = password
+                break
+    
+    account_dict[username] = account_info_dict
+
+    security_questions()
+
+    account_save()
+
+    return 
+
+# Asks user to pick a question and saves the answer
+def security_questions():
+     sec_questions_dict = {
+         "1":"What's your pet's name",
+         "2":"What's your mother's middle name",
+         "3":"What's your best friend's nickname",
+         "4":"Which hospital were you born in"
+      }
+
+     print("-"*26)
+     for key,val in sec_questions_dict.items():
+         print(key,":",val)
+     print("-"*26)
+
+     security_choice = input("Please choose a security question (Type in corresponding number):\n")
+
+     value = sec_questions_dict[security_choice]
+     print(value)
+     question_answer = input("")
+
+     account_info_dict["Security Question"] = value
+     account_info_dict["Security Answer"] = question_answer
+
+     return value, question_answer
 
 # Asks to generate a randomized password
 def password_randomizer_query():
+    accepted_answers = 'yes','no'
+
+    while True:
+        rand_answer = input("Do you want a randomized password? ('yes' or 'no'):\n")
+        if rand_answer == 'yes':
+            gen_pass = random_pass()
+            account_info_dict["Password"] = gen_pass
+            print("Your random pass is: " + gen_pass)
+            break
+        elif rand_answer == 'no':
+            break
+        elif rand_answer != accepted_answers:
+            print("Incorrect use.")
+
+    return rand_answer
 
-    return
 # Generates a randomized password using letters and digits
 def random_pass():
+    string = shane_char.ascii_characters_and_num()
+    letters_and_num = string.letters + string.digits
+    rand_password = random.choice(string.lowercase)
+    rand_password += random.choice(string.uppercase)
+    rand_password += random.choice(string.digits)
+    rand_password += random.choice(string.special_characters)
+
+    for characters in range(6):
+        rand_password += random.choice(letters_and_num)
```

Comment:
I think what you're trying to go for here is having a good spread of the different character types, but this is a lot of work to do that. 

All you need to really do is get a big list with all the different character types and then have the python random functions pick out as many as you want. That way you get a nice mix instead right off the bat instead of having to shuffle them.

In this solution they may not have all the different character types every time, but a good password is more about length than anything else.

Reply:
I think I understand what you mean.